### PR TITLE
feat: enable schema validation

### LINKED ISSUE #289: Enable schema validation in tdp-lib
Following https://github.com/TOSIT-IO/tdp-collection/pull/609, we need tdp-lib to be able to validate the schema, and expose the schema through its api.

### PR DESCRIPTION
**Pull request by gboutry**
#### Which issue(s) this PR fixes

Fixes #289

#### Additional comments

This PR adds schema validation to tdp-lib, and allows a collection to expose it's schema definitions.

Collections can extend an existing json schema, this will be treated as additional constraints. 
This means incompatibilities between 2 definitions of a same schema will raise an exception in validation stage.

Schema definition is allowed at service level, not component for the moment.

Requires:
- [x] https://github.com/TOSIT-IO/tdp-collection/pull/609

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp/core/collection.py [hunk 1]
```diff
@@ -3,10 +3,14 @@
 
 from pathlib import Path
 
+from networkx.readwrite.json_graph.jit import json
```
Comment (rpignolet) [inline tdp/core/collection.py:6]: Why not use `import json` ?

File: tdp/core/variables/cluster_variables.py [hunk 2]
```diff
@@ -27,7 +28,11 @@ def __iter__(self):
 
     @staticmethod
     def initialize_cluster_variables(
-        collections, tdp_vars, override_folder=None, repository_class=GitRepository
+        collections,
+        tdp_vars,
+        override_folder=None,
+        repository_class=GitRepository,
+        validate=False,
```
Comment (rpignolet) [inline tdp/core/variables/cluster_variables.py:35]: Why disable validation ?

Comment (gboutry) [reply tdp/core/variables/cluster_variables.py:35]: To make it opt-in, there might be use case where we define the schema wrongly, and we shouldn't prevent initialization is a tdp-lib user **realy** wants to initialize

File: tdp/core/collection.py (hunk 1 above)
Comment (gboutry) [reply tdp/core/collection.py]: Oups, automatic import from my ide, it's fixed

File: tdp/core/variables/cluster_variables.py (hunk 2 above)
Comment (rpignolet) [reply tdp/core/variables/cluster_variables.py:35]: Yes, and this is a default value, the CLI call it with `True` by default.

**Review by rpignolet — APPROVED** [review]

**Review by leopaul36 — APPROVED** [review]

**Pull request merged**